src/svgdom/util/elements_by_tag_name_finder.cpp

```diff
@@ -0,0 +1,115 @@
+#include "elements_by_tag_name_finder.hpp"
+
+#include <utki/debug.hpp>
+
+#include "../visitor.hpp"
+
+using namespace svgdom;
+
+namespace{
+class cache_creator : virtual public svgdom::const_visitor{
+public:
+	std::unordered_map<std::string, std::vector< const element*>> cache;
+
+	void add_to_cache(const svgdom::element& e, const svgdom::styleable& s){
+		if (!s.get_tag().empty()) {
+			auto it = cache.find(s.get_tag());
+
+			if (it != cache.end()) {
+				it->second.push_back(&e);
+			} else {
+				std::vector<const element*> elements = {&e};
+				cache.insert(std::make_pair(s.get_tag(), std::move(elements)));
+			}
+		}
+	}
+	
+	void visit_container(const svgdom::element& e, const svgdom::container& c, const svgdom::styleable& s){
+		this->add_to_cache(e, s);
```

Comment:
same thing here, we don't need `styleable` argument, the `get_tag()` method is now present in base `element` class.